### PR TITLE
Azure Key Vault TemplateSpec Solution-- DO NOT MERGE AS 1P

### PR DESCRIPTION
**Pull request by v-atulyadav**
Change(s):

See guidance below
Reason for Change(s):
This PR is part of template spec solution migration for solution Azure Key Vault.

See guidance below
Version Updated:
2.0.0

Required only for Detections/Analytic Rule templates
See guidance below
Testing Completed:
The link for test results is - https://microsoftapc.sharepoint.com/:w:/r/teams/SentinelCATTeam/_layouts/15/Doc.aspx?sourcedoc=%7BA5119E85-10AA-4F7B-A8C5-AAAF63E9699A%7D&file=Sentinel%20Solution%20Migration%20-TemplateSpec%20-%20Azure%20Key%20Vault.docx&action=default&mobileredirect=true

See guidance below
This is an 1P solution.

### CODE REVIEW

**Review by mkchiliveri — DISMISSED** [review]
Looks good, Approved.

**Comment by v-marimanda** [comment]
Hi @v-sabiraj
Can you please get confirmation on this PR merging or closing and act accordingly.
Thanks

**Comment by NikTripathi** [comment]
@v-atulyadav Please resolve conflicts.

File: Solutions/Azure Key Vault/Data Connectors/AzureKeyVault.JSON [hunk 1]
```diff
@@ -0,0 +1 @@
...signment","policyDefinitionGuid":"951af2fa-529b-416e-ab6e-066fd85ac459","assignMode":1},"type":"InstallAgent"}]}]}]},"type":"InstructionStepsGroup"}]}],"permissions":{"resourceProvider":[{"provider":"Microsoft.OperationalInsights/workspaces","permissionsDisplayText":"read and write permissions.","providerDisplayName":"Workspace","scope":"Workspace","requiredPermissions":{"read":true,"write":true,"delete":true}}],"customs":[{"name":"Policy​","description":"owner role assigned for each policy assignment scope.​"}]},"availability":{"status":2,"featureFlag":{"feature":"AzureKeyVaultFeature","featureStates":{"1":2,"2":2,"3":2,"4":2,"5":2,"6":1,"7":1,"8":2}}},"ascAssessment":{"assessmentKey":"b1af52e4-e968-4e2b-b6d0-6736c9651f0a","assessmentLinkText":"Enable Microsoft Defender for Key-Vaults now","assessmentDescriptions":"Microsoft Defender for Key-Vaults provides real-time protection for your Azure Key-Vaults Services. Enable Microsoft Defender for Key-Vaults,connect it into Sentinel, and use Key-Vaults diagnostic logs in order to investigate Microsoft Defender alerts into a root cause in Sentinel.","assessmentSectionTitleText":"Enable Microsoft Defender for Key-Vaults - Recommended!"}}
```
Comment (v-amolpatil) [inline Solutions/Azure Key Vault/Data Connectors/AzureKeyVault.JSON:1]: please format this file json it is coming in one line

**Review by v-sabiraj — APPROVED** [review]

**Review by v-dvedak — APPROVED** [review]

**Pull request merged**